gadget/install/install.go

```diff
@@ -624,3 +625,48 @@ func FactoryReset(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string,
 		DeviceForRole: deviceForRole,
 	}, nil
 }
+
+// MountVolumes mounts partitions for the volumes specified by
+// onVolumes. It returns the ESP partition and a function that needs
+// to be called for unmounting them.
+func MountVolumes(onVolumes map[string]*gadget.Volume) (espMntDir string, unmount func() error, err error) {
+	var mountPoints []string
+	numEsp := 0
+	unmount = func() (err error) {
+		for _, mntPt := range mountPoints {
+			errUnmount := sysUnmount(mntPt, 0)
+			if errUnmount != nil {
+				logger.Noticef("cannot unmount %q: %v", mntPt, errUnmount)
+			}
+			// Make sure we do not set err to nil if it had already an error
+			if errUnmount != nil {
+				err = errUnmount
+			}
+		}
+		return err
+	}
+	for _, vol := range onVolumes {
+		for _, part := range vol.Structure {
+			if part.Filesystem == "" {
+				continue
+			}
+			mntPt := filepath.Join(boot.InitramfsRunMntDir, part.Name)
+			if err := mountFilesystem(part.Device, part.Filesystem, mntPt); err != nil {
+				defer unmount()
+				return "", nil, fmt.Errorf("cannot mount %q at %q: %v", part.Device, mntPt, err)
+			}
+			mountPoints = append(mountPoints, mntPt)
+
+			if strings.Contains(strings.ToUpper(part.Type), "C12A7328-F81F-11D2-BA4B-00A0C93EC93B") {
```

Comment:
(nitpick) I wonder if we should have something like `ESPGUID = "C12A..."` somewhere but not really important, the context here is super clear.